### PR TITLE
Enhancement/Improve tslint setup for doc checking

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] ~I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.~
- [x] ~I documented my source code using the JavaDoc / JSDoc style.~
- [x] ~I added integration test cases for the server (Spring) related to the features~
- [x] ~I added integration test cases for the client (Jest) related to the features~
- [x] I added screenshots/screencast of my UI changes
- [x] ~I translated all the newly inserted strings~

### Motivation and Context

- Linting fails atm because of new rule for member-ordering
- Docs are not checked atm

### Description

Fixes 2 issues from https://github.com/ls1intum/Artemis/pull/739:
- Corrected the completed-docs rules (were not working because of missing array brackets)
- Removed a rule for member ordering we don't follow (I would prefer if we could take a look at other rules after fixing the docs first)

Also there is now a separate build step for checking the TS docs so we don't accidentally introduce new linting issues.

:warning: The longer we wait to merge this PR, the more likely it is that we introduce new linting issues with new merged PRs.

### Steps for Testing

1. Run `yarn lint` => No errors/warnings should appear in console
2. Run `yarn lint:doc` => Documentation errors should appear in console, process should fail

### Screenshots

image

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -45,16 +45,22 @@ jobs:
...
         script: yarn lint
+      - stage: Lint
+        name: "TypeScript Documentation"
+        language: node_js
+        node_js: 12.8.0
+        install: yarn install
+        script: yarn lint:doc
     allow_failures:
-      - script: yarn lint
+      - script: yarn lint-doc
```
Comment (thilo-behnke) [inline .travis.yml]:
```suggestion
      - script: yarn lint:doc
```

**Comment by jpbernius** [comment]
ℹ️ This would require an update of our Bamboo config also!

**Comment by krusche** [comment]
Why would you like to distinguish between `yarn lint` and `yarn lint:doc`?
Two commands make it only more difficult.

I think `yarn lint` should check everything if possible

**Comment by thilo-behnke** [comment]
> Why would you like to distinguish between `yarn lint` and `yarn lint:doc`?
> Two commands make it only more difficult.
> 
> I think `yarn lint` should check everything if possible

**1) It could happen that somebody introduces a new linting error that is unrelated to the doc check:**

```
var a = 4; // We don't allow var declarations
```
This issue would get lost in the log of missing doc checks and nobody would notice.

**2) The tslint.json is used by the IDE and `yarn start`**

So when compiling the client code locally or navigating through the files in an IDE, the doc errors would always show up. This is confusing and distracts from real issues.

**Pull request merged**